api/room/schema_query.py

```diff
@@ -256,6 +266,70 @@ class Query(graphene.ObjectType):
                 [required]"
     )
 
+    all_available_rooms = graphene.Field(
+        AvailableRooms,
+        start_date=graphene.String(required=True),
+        start_time=graphene.String(required=True),
+        end_date=graphene.String(required=True),
+        end_time=graphene.String(required=True),
+        time_zone=graphene.String(required=True),
+        location=graphene.String(required=True),
+        description="Returns available rooms for a specific time \
+            \n- start_time: Start time and date when you want to book room from\
+            [required]\n- end_time: time and date you want to book room upto\
+                [required]\n time_zone: The time zone of the location\
+                    [required]\n location: The location of the office's \
+                        room you want to book"
+    )
+
+    def resolve_all_available_rooms(self, info, **kwargs):
+
+        validate_date_input(kwargs['start_date'], kwargs['end_date'])
+        validate_date_input(kwargs['start_time'], kwargs['end_time'])
+
+        format_range_dates(kwargs['start_date'], kwargs['end_date'])
+
+        new_start_time = convert_date(kwargs['start_date'],
+                                      kwargs['start_time'],
+                                      kwargs['time_zone']
+                                      )
+        new_end_time = convert_date(kwargs['end_date'],
+                                    kwargs['end_time'],
```

Comment:
Hello @Ramadhan0,
I think you should validate start time and end time cause when start-time is greater than end-time it works.

Reply:
I did @mifeille

Reply:
cool